java/ql/lib/ext/java.util.model.yml

```diff
@@ -428,6 +430,7 @@ extensions:
       - ["java.util", "Collections", "emptySet", "()", "summary", "manual"]
       - ["java.util", "Collections", "sort", "", "summary", "manual"]
       - ["java.util", "Enumeration", "hasMoreElements", "()", "summary", "manual"]
+      - ["java.util", "EnumSet", "copyOf", "(EnumSet)", "summary", "manual"]
```

Comment:
I take it that this is safe because it can only contain elements from an enum already defined in the code. In that case should we make neutral models for all the other methods of `EnumSet` while we're at it?

Reply:
> this is safe because it can only contain elements from an enum already defined in the code

Yes, and I actually had doubts about it because we have models for the keys of an `EnumMap`, which should be a similar case: https://github.com/github/codeql/blob/main/java/ql/lib/ext/java.util.model.yml#L132-L135

But I can't think of how unrestricted user input could end up in there, so maybe we'd need to adjust those as well.

> should we make neutral models for all the other methods of EnumSet while we're at it?

That's another thing I struggled with in this PR: whether to aim for completeness when modeling things, or just cover what we know for sure is relevant. Nothing against modeling the other `EnumSet` methods, but this happens with other methods in this PR as well.

Reply:
Done in https://github.com/github/codeql/pull/15906/commits/36f6a6fb10cb0bdb04cc807675675f5257f0a554.

Reply:
If I have to understand a whole class to model one method then I normally write/improve the models for the rest of the class if it's easy, as I've already done the hard work. It seemed to me that the rationale for `EnumSet.copyOf` not propagating taint applies to the whole class, so it seems worth modelling the whole thing quickly, especially as you can copy the df-gen models so a lot of the work of listing all the methods is already done. But it's always a trade-off with not doing too much extra work beyond what is needed for the original method.